cmake/Platform/Core/BoardPropertiesReader.cmake

```diff
@@ -28,6 +28,34 @@ function(_GET_BOARD_ID BOARD_NAME BOARD_CPU TARGET_NAME OUTPUT_VAR)
     endif()
 endfunction()
 
+#=============================================================================#
+# _recursively_replace_properties
+# [PRIVATE/INTERNAL]
+#
+# _recursively_replace_properties(BOARD_ID PROPERTY_VALUE_VAR)
+#        BOARD_ID - return value from function "_get_board_id (BOARD_NAME, BOARD_CPU)". It contains BOARD_NAME and BOARD_CPU
+#        PROPERTY_VALUE_VAR - the value of a property that may contain
+#           references to other properties.
+#
+# Recursively replaces references to other properties.
+#
+#=============================================================================#
+function(_recursively_replace_properties BOARD_ID PROPERTY_VALUE_VAR)
+   while("${${PROPERTY_VALUE_VAR}}" MATCHES "[^\$]{([^}]*)}")
+      set(variable "${CMAKE_MATCH_1}")
+      if("${variable}" MATCHES "${BOARD_ID}\.([^}]*)")
```

Comment:
This second regex checks if the property (variable) that was referenced is one of  a board.